src/engine/SCons/Tool/wix.py

```diff
@@ -62,39 +64,58 @@ def generate(env):
 
     env['BUILDERS']['WiX'] = linker_builder
 
+
 def exists(env):
     env['WIXCANDLE'] = 'candle.exe'
     env['WIXLIGHT']  = 'light.exe'
 
-    # try to find the candle.exe and light.exe tools and 
-    # add the install directory to light libpath.
+    def check_wix(e, p):
+        """try to find the candle and light tools; add to path if found."""
+        try:
+            files = os.listdir(p)
+            if e['WIXCANDLE'] in files and e['WIXLIGHT'] in files:
+                e.PrependENVPath('PATH', p)
```

Comment:
I took the "exists" function to mean "I want to know if this thing currently exists in my environment" not "find and add it to my environment". Is there any documentation regarding the goal of the exists function?

Reply:
I can't find any - I was asking this not long ago on the list (seems everybody just writes a tool by stealing from another tool, which is fine).  Yes, I expect you're right about the intent, this is the resurrection of an old patch before I'd seen some of the recent discussion.  I'm still willing to add some verbiage to the docs on `generate`/`exists` if I figure out what we should say.  
The problem is... if we find the tool and so report (only), then somebody else has to repeat the work later to find it again, and that's hardly ideal either?  We had that issue with the java tests until things got more unified.